src/GafferSceneUI/TranslateTool.cpp

```diff
@@ -151,7 +159,56 @@ void TranslateTool::translate( const Imath::V3f &offset )
 	}
 }
 
-IECore::RunTimeTypedPtr TranslateTool::dragBegin()
+bool TranslateTool::keyPress( const GafferUI::KeyEvent &event )
+{
+	// We track this regardless of whether we're active or not in case the tool
```

Comment:
I'm not sure I understand this comment, or if I do, it isn't working. If the tool isn't active, I hold V, and while holding it, press W, the tool becomes active but without the pointer changing. I think I'd be happy with the simpler version where the tool has to be active already when I press V - does that cause problems of its own?

Reply:
@johnhaddon So currently, we do track the mode, so the tool _behaviour_ will be correct, but as you mentioned, the pointer doesn't get updated.

There is also a 🐞 that if you switch to a tool other than Translate/Rotate whilst `v` is held, you'll be stuck with the target cursor until someone else decides they have a cursor they'd rather show. That seems like a worse bug than the scenario of having to press `v` again if you held it before a compatible tool was activated. 

Seems like we should probably be watching for our active plug changing and wrangling the cursor there too?  It makes for more complexity - but I don't feel we should allow the sticky-cursor bug...

Reply:
Yeah - I hope all the signal ordering works in our favour :)

Reply:
Hopefully addressed in https://github.com/GafferHQ/gaffer/pull/3403/commits/25a892ed670f51f3a94fa629e534ff02c4b23a47